### ISSUE TITLE
1.16.5-1.19.2 - Major Issue - Unable to log back in after leaving server when the adaptive bundle is present

### ISSUE THREAD

**Issue by P3rf3ctXZer0** [MarkusBordihn/BOs-Adaptive-Performance-Tweaks#38]
**Describe the bug**
Okay so boy do I have a dozy for you; apparently when all the stars align and you add your adaptive bundle mod - it works great on a server exactly 1 time. Once you get disconnected you cannot reconnect. I honestly took forever to figure out why. It was this mod. I couldn't see how this was possible. Until now. I am 100% certain this is the cause. Somehow when using this mod and disconnect and reconnect the client appears to the server as vanilla. (I have no idea how it is doing this.) I have not pin pointed the exact cause so you might have to do some really hard digging yourself.

**To Reproduce**
Add your bundle to any major modpack including atm or anything that runs on 1.16.5+ Forge and has at least 100 mods.

**Expected behavior**
I mean the cause is ... well erroneous. I would expect your mod not make the client appear vanilla to the server so I guess that.

**Debugging**
[debug.log](https://github.com/MarkusBordihn/BOs-Adaptive-Performance-Tweaks/files/10833272/debug.log)
[latest.log](https://github.com/MarkusBordihn/BOs-Adaptive-Performance-Tweaks/files/10833273/latest.log)

**Is the issue happening in combination with other mods? (If yes, please provide a full list)**
Good question - I will need you to test but a quick test is add this to any atm or vault hunter pack.

**Additional context**
The exact time it takes to narrow down is sus also not sure if it is a mod thing or mod amount thing.

**Comment by P3rf3ctXZer0** [comment]
Its possible its because I also had them client but sometimes I do lan based testing so ... either way its an issue.

**Comment by P3rf3ctXZer0** [comment]
I already solved problem in part yet again XD narrowed it down to this 
![EOGYq6a 1](https://user-images.githubusercontent.com/10351221/221401460-3d46bad7-6545-4eee-9081-06d4d0cd4a56.png)

**Comment by MarkusBordihn** [comment]
Thanks for the report. Unfortunately I was not able to reproduce the issue with my reference mod pack like: https://www.curseforge.com/minecraft/modpacks/bos-adventure-world

Could you please confirm if you see the same issues with the reference mod pack on your side?

In general the mod has no connection to the login process / server query itself.
The server should not show up as "vanilla", which is clearly not cause by this mod.
The server will only show up as "vanilla" during the start up until all mods are loaded and processes, so only mods with mixins / core modding (client and server side) are really able to influence this after the start of the server.
None of my mods are using mixins / core modding and so they are not able to cause such effects.

I will run some tests (with the latest updates), but at the moment it's not looking like an issue with this mod.

**Comment by P3rf3ctXZer0** [comment]
I am getting sick or tired or both but if you want to diagnose issues just add it to my pack https://www.curseforge.com/minecraft/modpacks/fiction-friction-pixelmon/files/4413352 It has to be on a dedicated server. If it is not the bug does not trigger 100% of the time. More like 60/40 on lan.

**Comment by MarkusBordihn** [comment]
I hope you will get better soon. Unfortunately I was not able to reproduce the issue with my reference mod-pack and the "Adaptive Performance Tweak bundle".

I additional tried your mod-pack, which I was able to reproduce the issue but without the "adaptive bundle", see:
![image](https://user-images.githubusercontent.com/12183293/221442811-f8b9554e-e700-432c-90c7-a759bffb2b42.png)

First connect was fine, even if the client took ages to load.

Server shows me an "x" after I clicked "refresh", before I tried to connect a second time.

![image](https://user-images.githubusercontent.com/12183293/221442610-d4c47771-779c-43e5-9a30-718c6e5ecce6.png)

I tried to connect a second time but was unable to reconnect even after a wait time of 15min.

![image](https://user-images.githubusercontent.com/12183293/221442646-f451cc1f-f08d-40f2-93f2-fc8061c66c4a.png)

According the debug.log this was caused by the long loading times:
```
[26Feb.2023 23:54:21.115] [Server thread/INFO] [net.minecraft.server.network.ServerLoginPacketListenerImpl/]: Disconnecting com.mojang.authlib.GameProfile ... lost connection: Took too long to log in
[26Feb.2023 23:54:21.164] [Server thread/WARN] [net.minecraft.network.Connection/]: handleDisconnection() called twice
```

At the moment I see no direct issues with the adaptive bundle and more an issue with the mod pack itself.
